docs/proposals/keeping-transient-pod-name-spaces-alive.md

```diff
@@ -0,0 +1,54 @@
+# Keeping transient pod name spaces alive
+
+## Introduction
+The goal of this proposal is to give administrators options to keep namespaces and therefore APB pods after the execution. The use cases for this feature are demos and debugging APBs.
+
+## Problem Description
+The [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1497766) was created when we moved to creating transient namespaces during execution of the APB pod. What happens  This bug creates issues for debugging APB's as well as issues with demos. 
+
+## Implementation Details.
+
+### Configuration Values
+
+* `keep_namespace` or `ClusterConfig.KeepNamespace` - Parameter to always keep name space no matter what.
+* `keep_namespace_on_error` or `ClusterConfig.KeepNamespaceOnError` -  parameter to keep name space around if an error occurs in the play book that is running.
```

Comment:
Have we considered a simple TTL? Keep pods alive for x amount of time, then possibly have a reaper job to clean things up. The TTL, say 1 day, would allow customer support to debug a pod that had an issue.

Reply:
When I implemented the original sandboxes, there was an urgency around: the sandbox MUST be deleted following an APB method execution. I think there was a security concern with keeping resources like ServiceAccounts and RoleBindings around longer than required. We could keep the pod around and delete the SA/RB, but that might violate the idea of the atomic "sandbox" (deleting some resources, but not others). Maybe it's okay because the Pod itself isn't considered part of it, but those are my concerns with a TTL.

Reply:
Maybe the TTL is a 3rd configuration option?

Reply:
I think this a great idea, but it strikes me as a feature rather than a bug fix.